File: ARCHISOFT_topoalign/frmAlignTopo.vb
Public Class frmAlignTopo

    Dim OldVal As Boolean

    Private Sub btnAlign_Click(sender As Object, e As EventArgs) Handles btnAlign.Click
        Me.DialogResult = System.Windows.Forms.DialogResult.OK
        Me.Close()
    End Sub


    Private Sub rdoElem_CheckedChanged(sender As Object, e As EventArgs) Handles rdoElem.CheckedChanged
        If Me.rdoElem.Checked = True Then
            Me.gbFaceToUse.Enabled = True
        Else
            Me.gbFaceToUse.Enabled = False
        End If
    End Sub

    Private Sub rdoEdge_CheckedChanged(sender As Object, e As EventArgs) Handles rdoEdge.CheckedChanged

        If Me.rdoEdge.Checked = True Then
            OldVal = Me.chkRemoveInside.Checked
            Me.chkRemoveInside.Checked = False
            Me.chkRemoveInside.Enabled = False
        Else
            Me.chkRemoveInside.Enabled = True
            Me.chkRemoveInside.Checked = OldVal
        End If
    End Sub

    Private Sub frmAlignTopo_Load(sender As Object, e As EventArgs) Handles MyBase.Load

    End Sub

    Private Sub DisplayUnitcomboBox_SelectedIndexChanged(sender As Object, e As EventArgs) Handles DisplayUnitcomboBox.SelectedIndexChanged
        Me.DisplayUnitTypecomboBox.SelectedIndex = Me.DisplayUnitcomboBox.SelectedIndex
    End Sub
End Class

File: ARCHISOFT_topoalign/frmAlignTopo.cs
using System;

namespace ARCHISOFT_topoalign
{
    public partial class frmAlignTopo
    {
        public frmAlignTopo()
        {
            InitializeComponent();
            _btnAlign.Name = "btnAlign";
            _rdoEdge.Name = "rdoEdge";
            _rdoElem.Name = "rdoElem";
            _DisplayUnitcomboBox.Name = "DisplayUnitcomboBox";
        }

        private bool OldVal;

        private void btnAlign_Click(object sender, EventArgs e)
        {
            DialogResult = System.Windows.Forms.DialogResult.OK;
            Close();
        }

        private void rdoElem_CheckedChanged(object sender, EventArgs e)
        {
            if (rdoElem.Checked == true)
            {
                gbFaceToUse.Enabled = true;
            }
            else
            {
                gbFaceToUse.Enabled = false;
            }
        }

        private void rdoEdge_CheckedChanged(object sender, EventArgs e)
        {
            if (rdoEdge.Checked == true)
            {
                OldVal = chkRemoveInside.Checked;
                chkRemoveInside.Checked = false;
                chkRemoveInside.Enabled = false;
            }
            else
            {
                chkRemoveInside.Enabled = true;
                chkRemoveInside.Checked = OldVal;
            }
        }

        private void frmAlignTopo_Load(object sender, EventArgs e)
        {
        }

        private void DisplayUnitcomboBox_SelectedIndexChanged(object sender, EventArgs e)
        {
            DisplayUnitTypecomboBox.SelectedIndex = DisplayUnitcomboBox.SelectedIndex;
        }
    }
}